tests/python/pants_test/backend/native/tasks/test_c_compile.py

```diff
@@ -38,6 +54,21 @@ def create_simple_c_library(self, **kwargs):
                             sources=['test.h', 'test.c'],
                             **kwargs)
 
+  def test_header_only_noop_with_alternate_header_extension(self):
+    alternate_extension = '.asdf'
+    c = self.create_header_only_alternate_c_library(alternate_extension)
+    context = self.prepare_context_for_compile(target_roots=[c], options={
+      'c-compile-settings': {
+        'header_file_extensions': [alternate_extension],
+      },
+    })
+    c_compile = self.create_task(context)
+
+    with self.captured_logging(level=logging.INFO) as logs:
+      c_compile.execute()
+      info = list(logs.infos())[-1]
```

Comment:
OK - the tests that scrape logs - I can see maybe testing for warning or error logs since they are intended to always message the user, but info is info - its throwaway - or so you'd think until you edit the text and a test goes boom. Stringly consider killing this log test and instead testing the real deal. Prove the `ObjectFiles` product is empty.

Ditto in `test_cpp_compile.py` below.

Reply:
That's an extremely reasonable and easy fix.

Reply:
Removed both the logs (now debugs) and any checking for those logs (instead proving the `ObjectFiles` product is empty).